Source/ProgressionSystemRuntime/Public/Widgets/PSMenuWidget.h

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Blueprint/UserWidget.h"
+#include "PoolManagerTypes.h"
+#include "PSMenuWidget.generated.h"
+
+/**
+ * Widget to display the progression as stars in the main menu.
+ */
+UCLASS()
+class PROGRESSIONSYSTEMRUNTIME_API UPSMenuWidget : public UUserWidget
+{
+	GENERATED_BODY()
+
+public:
+	/**
+	 * Dynamically populates a Horizontal Box with images representing unlocked and locked progression icons.
+	 * @param AmountOfUnlockedPoints The number of images (unlocked-icon as images) to be displayed 
+	 * @param AmountOfLockedPoints The number of images (locked-icon as images) to be displayed
+	 */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void AddImagesToHorizontalBox(float AmountOfUnlockedPoints, float AmountOfLockedPoints);
+
+	/**
+	 * Dynamically populates a Horizontal Box with images representing unlocked and locked progression icons.
+	* @param CreatedObjects - Handles of objects from Pool Manager
+	* @param AmountOfUnlockedPoints The number of images (unlocked-icon as images) to be displayed 
+	 * @param AmountOfLockedPoints The number of images (locked-icon as images) to be displayed
+	 */
+	UFUNCTION(BlueprintCallable, Category= "C++")
+	void OnTakeFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects, float AmountOfUnlockedPoints, float AmountOfLockedPoints);
+
+	/**
+	* Sets the visibility of the background overlay and lock icon.
+	* @param VisibilitySlate The visibility state (e.g., Visible, Collapsed) to apply to the overlay and icon.
+	*/
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected, BindWidget))
+	void SetOverlayVisibility(ESlateVisibility VisibilitySlate);
```

Comment:
You have `BindWidget` specified in `UFUNCTION` (not property, but function).

`BindWidget` is specified used only for subwidget properties. Remove it from your function

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/52882081ac940815290c4c02252d5ff792c08da0